.github/workflows/e2e.yml

```diff
@@ -0,0 +1,102 @@
+name: "Test: End to End"
+
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - '**.md'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+
+jobs:
+  end-to-end-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Workspace prefix
+        id: prefix
+        run: echo "::set-output name=name::action-e2e-test-$(date +%s)"
+      
+      - name: Workspace names
+        id: workspaces
+        env:
+          workspaces: '["alpha", "beta"]'
+        run: echo "::set-output name=names::${{ toJSON(env.workspaces) }}"
+
+      - name: Run action
+        uses: ./
+        with:
+          runner_terraform_version: '1.0.5'
+          terraform_organization: takescoop-oss
+          terraform_token: ${{ secrets.TF_TOKEN_OSS }}
+          name: ${{ steps.prefix.outputs.name }}
+          tags: |-
+            - terraform:true
+          workspace_tags: |-
+            alpha:
+              - workspace:alpha
+            beta:
+              - workspace:beta
+          apply: true
+          variables: |-
+            - key: foo
+              value: bar
+              category: terraform
+          workspace_variables: |-
+            alpha:
+              - key: workspace
+                value: alpha
+                category: terraform
+            beta:
+              - key: workspace
+                value: beta
+                category: terraform
+          workspaces: ${{ steps.workspaces.outputs.names }}
+          backend_config: |-
+            local:
+              path: /github/workspace/terraform.tfstate
+
+      - name: Assertions
+        env:
+          TF_TOKEN: ${{ secrets.TF_TOKEN_OSS }}
+        run: |
+          statefile=$(cat terraform.tfstate) && echo "Success: state file present"
+
+          echo "::group::tfstate-file"
+          echo "$statefile"
+          echo "::endgroup::"
+
+          echo "::group::workspaces"
+
+          echo '${{ steps.workspaces.outputs.names }}' | jq -r '.[]' | while read -r ws ; do
+            workspace_url="https://app.terraform.io/api/v2/organizations/takescoop-oss/workspaces/${{ steps.prefix.outputs.name }}-$ws"
+
+            echo "GET $workspace_url"
+
+            curl \
+              --silent \
```

Comment:
removed `--fail` because I think we get more info without it that could be useful for debugging

```
       -f, --fail
              (HTTP)  Fail  silently  (no  output  at  all) on server errors. This is mostly done to better enable
              scripts etc to better deal with failed attempts. In normal  cases  when  an  HTTP  server  fails  to
              deliver  a  document,  it  returns  an  HTML document stating so (which often also describes why and
              more). This flag will prevent curl from outputting that and return error 22.

              This method is not fail-safe and there are occasions where non-successful response codes  will  slip
              through, especially when authentication is involved (response codes 401 and 407)
```

Reply:
Still needs to handle 4xx/5xx responses though:

https://daniel.haxx.se/blog/2021/02/11/curl-fail-with-body/

Reply:
ahh i see, without --fail (or --fail-with-body) we don't get the non-zero exit code. cool, will update.

Reply:
looks like we have an older version of curl [without --fail-with-body](https://github.com/TakeScoop/terraform-cloud-workspace-action/runs/3604220181?check_suite_focus=true). `--fail` isn't terrible, it's why I added the print URL line above if the workspace does not exist. We still get `curl: (22) The requested URL returned error: 404 ` which along with that should be enough to debug.